senders/discord/init.go

```diff
@@ -0,0 +1,98 @@
+package discord
+
+import (
+	"fmt"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/bwmarrin/discordgo"
+	"github.com/moira-alert/moira"
+)
+
+const messenger = "discord"
+
+// Sender implements moira sender interface for discord
+type Sender struct {
+	DataBase moira.Database
+	logger   moira.Logger
+	location *time.Location
+	session  *discordgo.Session
+	frontURI string
+}
+
+// Init reads the yaml config
+func (sender *Sender) Init(senderSettings map[string]string, logger moira.Logger, location *time.Location, dateTimeFormat string) error {
+	var err error
+	token := senderSettings["token"]
+	if token == "" {
+		return fmt.Errorf("cannot read the discord token from the config")
+	}
+	sender.session, err = discordgo.New("Bot " + token)
+	if err != nil {
+		return fmt.Errorf("error creating discord session: %s", err)
+	}
+	sender.logger = logger
+	sender.frontURI = senderSettings["front_uri"]
+	sender.location = location
+
+	handleMsg := func(s *discordgo.Session, m *discordgo.MessageCreate) {
+		if err := sender.getResponse(s, m); err != nil {
+			sender.logger.Errorf("failed to handle incoming message: %s", err)
+		}
+	}
+	sender.session.AddHandler(handleMsg)
+
+	go sender.runBot()
+	return nil
+}
+
+func (sender *Sender) getResponse(s *discordgo.Session, m *discordgo.MessageCreate) error {
```

Comment:
Are you really sure that discord needs the same logic with the bot as telegram? Telegram has stupid restrictions about user authorization (bot cannot send message to you if you didn't use !start command). I think you must check that discord has the same restrictions (or some other?)

Reply:
The !start command is being used just to associate the username/channel name with the unique channel ID. In the case of an individual user, any message would work but I used "!start" just like in telegram and discord channel to ensure consistency. Once the start command is used the user can use the username like "@aswinmprabhu" in the web UI instead of the channel ID. So I think it is necessary for discord as well but for a different reason.

Reply:
But we can still use channel(user) ID as a unique ID for contact? Interesting, what does discord suggest doing in that cases. Hmmm.

Reply:
There is some opportunity to see channel(user) ID from discord interface?

Reply:
Yes, channel ID is the only way to uniquely identify a channel (both text and private). As mentioned in [this comment](https://github.com/moira-alert/moira/pull/387#discussion_r298567230),
> [This](https://support.discordapp.com/hc/en-us/articles/206346498-Where-can-I-find-my-User-Server-Message-ID-) is the official article from discord support on how to get a channel ID. It also mentions how to get the channel ID of a private chat with an individual user. 
But @kamaev suggested switching to a telegram like solution that will enable users to just use the channel name or username in the web UI. This is more user-friendly but does involve issuing commands like "!start" to get it to work.